lib/cn/csched_sp3_work.c

```diff
@@ -94,20 +94,10 @@ sp3_work_estimate(struct cn_compaction_work *w, uint internal_children, uint lea
             break;
 
         case CN_ACTION_SPILL:
-            /* If any child is an internal node, then assume
-             * this operation will simply move data to other internal
-             * nodes (no net effect on samp).  Otherwise assume all data
-             * will move to leaf nodes.
-             *
-             * If we have both leaf and internal children, then data will
-             * be split between 'i_alen' and 'l_alen', but this usually
-             * happens in prefix trees where one prefix dominates, in
-             * which case most of the data will land in the internal
-             * node used by the dominant prefix.
-             */
+            assert(cn_node_isroot(w->cw_node));
```

Comment:
What should we do in a release build?

Reply:
You could ask that about every assert in our system right?  I mentioned in another comment that a bulletproof approach would be to back up asserts with errors and handle them.  But I don't think it's a good idea to start down that path without considering the impact.

Reply:
Right, at some point we gotta trust our chops and testing, at least a little :-)